source/base/parameter_handler.cc

```diff
@@ -762,6 +762,43 @@ ParameterHandler::parse_input_from_json(std::istream &in,
 
 
 
+void
+ParameterHandler::create_default_input_file(const std::string &filename,
```

Comment:
Same here: this should take an `OutputStyle`

Reply:
When using OutputStyle as parameter, it is difficult to use this function. The aim of this function is to be able to create a default input file in case none is existing so far (meaning that you can also not configure OutputStyle from parsed parameters). All one has is the filename, and the aim of this function is that the filename ending decides in which format to write the parameter file. However, ParameterHandler has to check the correctness of OutputStyle, which means that the user has to configure OutputStyle correctly before calling this function. Now, to avoid if-else-branches in the application code (which is the purpose of this function), I believe it is necessary and practical to change back to a minimal interface.

Reply:
In my opinion, the second half of ``ParameterAcceptor::initialize()`` (which I did not touch so far) suffers also from this design problem. The "prm" branch currently has an Assert

https://www.dealii.org/developer/doxygen/deal.II/parameter__acceptor_8cc_source.html

 but similar Asserts would be necessary for the other branches. I add these Asserts for this PR but my recommendation is to avoid the redundancy (output format is contained both in ending of filename and in ``OutputStyle``) and change this in a follow-up PR. For example, the "json" version is also not supported by the second half of  ``ParameterAcceptor::initialize()``.